IngestDocument is sending all the document in 1 request

### Describe the issue

If I have a lot of document in my rag, using `IngestDocument` will send all the document inside a single request.

We need to add a `bulkSize` (name TDB) parameters (default 500) that will segment the list of docs. 

Example error with elastic embedding store: 

```
2026-01-28 23:20:32.883dev.langchain4j.store.embedding.elasticsearch.ElasticsearchRequestFailedException: method [POST], host [https://elasticsearch.xxx.kestra.io],/ URI [/_bulk], status line [HTTP/1.1 413 Request Entity Too Large]
at dev.langchain4j.store.embedding.elasticsearch.ElasticsearchEmbeddingStore.addAll(ElasticsearchEmbeddingStore.java:348)
	at dev.langchain4j.store.embedding.EmbeddingStore.addAll(EmbeddingStore.java:66)
	at dev.langchain4j.store.embedding.EmbeddingStoreIngestor.ingest(EmbeddingStoreIngestor.java:202)
	at io.kestra.plugin.ai.rag.IngestDocument.run(IngestDocument.java:185)
	at io.kestra.plugin.ai.rag.IngestDocument.run(IngestDocument.java:37)
	at io.kestra.worker.WorkerTaskCallable.doCall(WorkerTaskCallable.java:81)
	at io.kestra.worker.AbstractWorkerCallable.call(AbstractWorkerCallable.java:64)
	at io.kestra.ee.worker.WorkerSecurityService.lambda$callInSecurityContext$0(WorkerSecurityService.java:30)
	at io.kestra.ee.sandbox.SecurityContextHolder.callInSecurityContext(SecurityContextHolder.java:22)
	at io.kestra.ee.worker.WorkerSecurityService.callInSecurityContext(WorkerSecurityService.java:28)
	at io.kestra.worker.DefaultWorker.lambda$callJob$18(DefaultWorker.java:981)
	at io.kestra.core.trace.NoopTracer.inCurrentContext(NoopTracer.java:15)
	at io.kestra.worker.DefaultWorker.callJob(DefaultWorker.java:977)
	at io.kestra.worker.DefaultWorker.runAttempt(DefaultWorker.java:926)
	at io.kestra.worker.DefaultWorker.run(DefaultWorker.java:720)
	at io.kestra.worker.DefaultWorker.handleTask(DefaultWorker.java:411)
	at io.kestra.worker.DefaultWorker.lambda$run$8(DefaultWorker.java:289)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.elasticsearch.client.ResponseException: method [POST], host [https://elasticsearch.xxx.kestra.io],/ URI [/_bulk], status line [HTTP/1.1 413 Request Entity Too Large]
at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:351)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:317)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
	at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:92)
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:138)
	at co.elastic.clients.elasticsearch.ElasticsearchClient.bulk(ElasticsearchClient.java:531)
	at dev.langchain4j.store.embedding.elasticsearch.ElasticsearchEmbeddingStore.bulkIndex(ElasticsearchEmbeddingStore.java:369)
	at dev.langchain4j.store.embedding.elasticsearch.ElasticsearchEmbeddingStore.addAll(ElasticsearchEmbeddingStore.java:346)
	... 20 more
```

### Environment

- Kestra Version: develop

--- a/src/main/java/io/kestra/plugin/ai/rag/IngestDocument.java
+++ b/src/main/java/io/kestra/plugin/ai/rag/IngestDocument.java
@@ -21,6 +21,7 @@ import io.kestra.core.utils.ListUtils;
 import io.kestra.plugin.ai.domain.EmbeddingStoreProvider;
 import io.kestra.plugin.ai.domain.ModelProvider;
 import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.Min;
 import jakarta.validation.constraints.NotNull;
 import lombok.*;
 import lombok.experimental.SuperBuilder;
@@ -140,6 +141,13 @@ public class IngestDocument extends Task implements RunnableTask<IngestDocument.
     @Builder.Default
     private Property<Boolean> drop = Property.ofValue(Boolean.FALSE);
 
+    @Schema(
+        title = "Bulk ingestion size",
+        description = "Maximum number of documents sent per ingestion request."
+    )
+    @Builder.Default
+    private Property<@Min(1) Integer> bulkSize = Property.ofValue(500);
+
     @Override
     public Output run(RunContext runContext) throws Exception {
         List<Document> documents = new ArrayList<>();
@@ -182,30 +190,46 @@ public class IngestDocument extends Task implements RunnableTask<IngestDocument.
         }
 
         EmbeddingStoreIngestor ingestor = builder.build();
-        IngestionResult result = ingestor.ingest(documents);
+        int rBulkSize = runContext.render(bulkSize).as(Integer.class).orElse(500);
+
+        Integer inputTokenCount = null;
+        Integer outputTokenCount = null;
+        Integer totalTokenCount = null;
+
+        for (int i = 0; i < documents.size(); i += rBulkSize) {
+            List<Document> batch = documents.subList(i, Math.min(i + rBulkSize, documents.size()));
+            IngestionResult result = ingestor.ingest(batch);
+
+            if (result.tokenUsage() != null) {
+                if (result.tokenUsage().inputTokenCount() != null) {
+                    inputTokenCount = (inputTokenCount == null ? 0 : inputTokenCount) + result.tokenUsage().inputTokenCount();
+                }
+                if (result.tokenUsage().outputTokenCount() != null) {
+                    outputTokenCount = (outputTokenCount == null ? 0 : outputTokenCount) + result.tokenUsage().outputTokenCount();
+                }
+                if (result.tokenUsage().totalTokenCount() != null) {
+                    totalTokenCount = (totalTokenCount == null ? 0 : totalTokenCount) + result.tokenUsage().totalTokenCount();
+                }
+            }
+        }
 
         runContext.metric(Counter.of("indexed.documents", documents.size()));
-        if (result.tokenUsage() != null) {
-            if (result.tokenUsage().inputTokenCount() != null) {
-                runContext.metric(Counter.of("input.token.count", result.tokenUsage().inputTokenCount()));
-            }
-            if (result.tokenUsage().outputTokenCount() != null) {
-                runContext.metric(Counter.of("output.token.count", result.tokenUsage().outputTokenCount()));
-            }
-            if (result.tokenUsage().totalTokenCount() != null) {
-                runContext.metric(Counter.of("total.token.count", result.tokenUsage().totalTokenCount()));
-            }
+        if (inputTokenCount != null) {
+            runContext.metric(Counter.of("input.token.count", inputTokenCount));
+        }
+        if (outputTokenCount != null) {
+            runContext.metric(Counter.of("output.token.count", outputTokenCount));
+        }
+        if (totalTokenCount != null) {
+            runContext.metric(Counter.of("total.token.count", totalTokenCount));
         }
 
         var output = Output.builder()
             .ingestedDocuments(documents.size())
-            .embeddingStoreOutputs(embeddings.outputs(runContext));
-
-        if (result.tokenUsage() != null) {
-            output = output.inputTokenCount(result.tokenUsage().inputTokenCount())
-                .outputTokenCount(result.tokenUsage().outputTokenCount())
-                .totalTokenCount(result.tokenUsage().totalTokenCount());
-        }
+            .embeddingStoreOutputs(embeddings.outputs(runContext))
+            .inputTokenCount(inputTokenCount)
+            .outputTokenCount(outputTokenCount)
+            .totalTokenCount(totalTokenCount);
 
         return output.build();
     }

--- a/src/test/java/io/kestra/plugin/ai/rag/IngestDocumentTest.java
+++ b/src/test/java/io/kestra/plugin/ai/rag/IngestDocumentTest.java
@@ -23,6 +23,7 @@ import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.IntStream;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
@@ -171,8 +172,7 @@ class IngestDocumentTest extends ContainerTest {
         int bulkSize = 5;
         int totalDocs = 12;
 
-        List<IngestDocument.InlineDocument> docs =
-            java.util.stream.IntStream.range(0, totalDocs)
+        List<IngestDocument.InlineDocument> docs = IntStream.range(0, totalDocs)
                 .mapToObj(i -> IngestDocument.InlineDocument.builder()
                     .content(Property.ofValue("doc-" + i))
                     .build()

--- a/src/test/java/io/kestra/plugin/ai/rag/IngestDocumentTest.java
+++ b/src/test/java/io/kestra/plugin/ai/rag/IngestDocumentTest.java
@@ -161,6 +161,48 @@ class IngestDocumentTest extends ContainerTest {
         assertKvStore(kvStore, kvKey, 2);
     }
 
+    @Test
+    void inlineDocumentsWithBulkSize() throws Exception {
+        RunContext runContext = runContextFactory.of("namespace", Map.of(
+            "modelName", "tinydolphin",
+            "endpoint", ollamaEndpoint
+        ));
+
+        int bulkSize = 5;
+        int totalDocs = 12;
+
+        List<IngestDocument.InlineDocument> docs =
+            java.util.stream.IntStream.range(0, totalDocs)
+                .mapToObj(i -> IngestDocument.InlineDocument.builder()
+                    .content(Property.ofValue("doc-" + i))
+                    .build()
+                )
+                .toList();
+
+        var task = IngestDocument.builder()
+            .provider(
+                Ollama.builder()
+                    .type(Ollama.class.getName())
+                    .modelName(Property.ofExpression("{{ modelName }}"))
+                    .endpoint(Property.ofExpression("{{ endpoint }}"))
+                    .build()
+            )
+            .embeddings(KestraKVStore.builder().build())
+            .drop(Property.ofValue(true))
+            .bulkSize(Property.ofValue(bulkSize))
+            .fromDocuments(docs)
+            .build();
+
+        IngestDocument.Output output = task.run(runContext);
+
+        assertThat(output.getIngestedDocuments()).isEqualTo(totalDocs);
+
+        String kvKey = (String) output.getEmbeddingStoreOutputs().get("kvName");
+        KVStore kvStore = runContext.namespaceKv(runContext.flowInfo().namespace());
+        assertKvStore(kvStore, kvKey, totalDocs);
+    }
+
+
     private void assertKvStore(KVStore kvStore, String kvKey, int nbDocuments) throws IOException, ResourceExpiredException {
         Optional<KVEntry> kvEntry = kvStore.get(kvKey);
         assertThat(kvEntry.isPresent()).isTrue();

